public/src/scenes/mainTask.js

```diff
@@ -922,3 +801,127 @@ var showBottomScreenPanel = function(context, titleText, subtitleText, bottomBut
+            // if threshold is greater than the original maxPress: thresholdMax is updated 
+            if (threshold > maxPressCount) {
+                thresholdMax = threshold
+                var recalibration = 1;
+            }
+            else {
+                // continue with thresholdMax at maxPressCount 
+                var recalibration = 0;
+                thresholdMax = maxPressCount;
+            }
+        }
+        else {// the trial wasn't successful or did not need recalibration: 
+            var recalibration = 0; // record recalibration didn't occur
+            // also keep thresholdMax at maxPressCount
+            thresholdMax = maxPressCount
+
+        }
+        // save thresholdMax
+        context.registry.set("thresholdMax", { thresholdMax });
+        // save it in its own document for easy retrieval later 
+        // saveThresholdMax(this.registry.get("thresholdMax"));        // [for firebase]
+    }
+    else { // if we are past the first calibration trials 
+        var recalibration = 0; // record recalibration didn't occur
+        thresholdMax = maxPressCount // do not adjust thresholdMax 
+    };
+
+    if (choice == 'timeout') {
+        // if the choice was a timeout then reset all the relevant variables so the payload doesn't retain the previous trial's data
+        trialEffortPropChosen = 0;
+        trialEffort = 0;
+        choiceRT = 0;
+        pressCount = 0;
+        pressTimes = [];
+        trialSuccess = 0;
+        pressStartTime = 0;
+        pressEndTime = 0;
+        coinsWonThisTrial = 0;
+    }
+
+    // set data to be saved into registry
+    let taskAttempt = new TaskAttempt(
+        trialNo,
+        trialStartTime,
+        trialReward1,
+        trialEffort1,
+        trialEffortPropMax1,
+        trialReward2,
+        trialEffort2,
+        trialEffortPropMax2,
+        choice,
+        choiceRT,
+        pressCount,
+        pressTimes,
+        trialSuccess,
+        nCoins,
+        trialEndTime,
+        effortTime,
+        recalibration,
+        thresholdMax
+    );
+
+    // save the data in a registry for later retrieval
+    context.registry.set("trial" + trialNo, taskAttempt);
+
+    // save data
+    EmbedContext.sendMessage("trialResult", taskAttempt.stringify());
+    console.log(context.registry.get("trial" + trialNo));
+    // saveTaskData(trial, this.registry.get(`trial${trial}`));        // [for firebase]
+    //saveTrialDataPav(this.registry.get(`trial${trial}`));         // [for Pavlovia deployment only]
+
+    // as a fallback for the case where the player misses the coins, we will add the coins regardless if the player touches them or not
+    if (trialSuccess && choice == 'route 1') {
+        coinsWonThisTrial = trialReward1;
+        nCoins += coinsWonThisTrial;
+    } else if (trialSuccess && choice == 'route 2') {
+        coinsWonThisTrial = trialReward2;
+        nCoins += coinsWonThisTrial;
+    } else {
+        coinsWonThisTrial = 0;
+    }
```

Comment:
Should this be done before we create and save the `taskAttempt` object? To me this seems like `nCoins` would be wrong.